osu.Game.Rulesets.Catch/Skinning/Legacy/CatchLegacySkinTransformer.cs

```diff
@@ -103,6 +103,16 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
 
                     result.Value = LegacyColourCompatibility.DisallowZeroAlpha(result.Value);
                     return (IBindable<TValue>)result;
+
+                case CatchSkinConfiguration config:
+                    switch (config)
+                    {
+                        case CatchSkinConfiguration.FlipCatcherPlate:
+                            // Always return `false` (don't flip catcher plate contents) for a legacy skin.
+                            return (IBindable<TValue>)new Bindable<bool>();
```

Comment:
The way this is currently done means that the flipping will not take place on the default skin either if beatmap skins are turned on.

Unsure how to resolve this but seems like unintentional behaviour.

Reply:
It may seem like an unintentional behaviour, but the eventual plan for legacy beatmap skins is to be backed by a default legacy skin if the user skin is not legacy, so it makes some sort of sense for this to remain here, I think.

Reply:
To fix that, this should probably have a pre-check on whether a catcher is provided, and return null if it isn't.

```csharp
if (this.GetDrawableComponent(Catcher) == null)
    return null;
```